runner/src/simulate.rs

```diff
@@ -700,21 +730,33 @@ impl ScenarioState {
         let mut pending_count = 0;
         let mut failed_count = 0;
         let mut not_requested_count = 0;
-        let wait_duration = Duration::from_secs(60 * 60);
-        // TODO_3164_1 : Make this a parameter, pass it in from the scenario config
-        // TODO_3164_3 : Code clean-up : Move redis calls to separate file move it out of simulate.rs
-        // TODO_3164_4 : Code clean-up : Move api call (fetch stream) to model_instance.rs?, maybe rename it
+        // TODO_AES_84_1: Validate sync between peers
+        // TODO_AES_84_2: Make this a nightly perf test. With a green signal when bith the validations pass
+        info!(
+            "Anchor wait time: {}",
+            self.scenario_opts.anchor_wait_time.unwrap()
+        );
+        let wait_duration =
+            Duration::from_secs(self.scenario_opts.anchor_wait_time.unwrap_or_default() as u64);
         sleep(wait_duration).await;
-
-        // Pick a peer at random
+        info!("Waiting for {} seconds", wait_duration.as_secs());
+        // Pick the first peer as the validator. It should have all the streams synced to it
         let peer = self.peers.first().unwrap();
 
         let ids = self.get_set_from_redis(ANCHOR_REQUEST_MIDS_KEY).await?;
         info!("Number of MIDs: {}", ids.len());
 
+        // Calculate the sample size
+        let sample_size = calculate_sample_size(ids.len(), 0.99, 0.02).await;
+        info!("Sample size: {}", sample_size);
+
+        // Select a random sample set of IDs
+        let sample_ids = select_sample_set_ids(&ids, sample_size).await;
+        info!("Sampled IDs count: {}", sample_ids.len());
+
         // Make an API call to get the status of request from the chosen peer
-        for stream_id in ids.clone() {
-            info!("Fetching anchor status for streamID {}", stream_id);
+        for stream_id in sample_ids.iter() {
+            // Only fetch anchor status for every thousandth stream ID
```

Comment:
I don't think this comment is accurate with the stats stuff in the mix

Reply:
It is not, will remove it